src/Service/Taf/NoaaService.php

```diff
@@ -0,0 +1,109 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Service\Taf;
+
+use App\Traits\ZMQContextTrait;
+use Cake\Console\ConsoleIo;
+use Cake\Datasource\ModelAwareTrait;
+use Cake\Http\Client;
+
+/**
+ * @property \App\Model\Table\AirportsTable $Airports
+ * @property \App\Model\Table\TafTable $Taf
+ */
+class NoaaService
+{
+    use ModelAwareTrait;
+    use ZMQContextTrait;
+
+    /**
+     * @var string
+     */
+    protected const _NOAA_URL = 'https://aviationweather.gov/adds/dataserver_current/httpparam?dataSource=tafs&requestType=retrieve&format=xml&mostRecentForEachStation=true&hoursBeforeNow=0&stationString=';
+
+    /**
+     * @var \Cake\Console\ConsoleIo|null
+     */
+    protected ?ConsoleIo $_io;
+
+    /**
+     * @var string|null
+     */
+    protected ?string $_rawTaf = null;
+
+    /**
+     * @var \Cake\Http\Client
+     */
+    protected Client $_client;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->loadModel('Airports');
+        $this->loadModel('Taf');
+
+        $this->_client = new Client();
+    }
+
+    /**
+     * @return void
+     */
+    public function getTaf(): void
+    {
+        $airports = $this->Airports->find()
+            ->select(['icao'])
+            ->enableHydration(false)
+            ->all()
+            ->extract('icao')
+            ->toArray();
+
+        $rawTaf = $this->_fetchTaf($airports);
```

Comment:
Minor thing, but we could spare the request if we have no airports (which is unlikely ofc 🙂)